common/src/Microsoft.Azure.IIoT.Hub.Module.Client/src/Default/IoTSdkFactory.cs

```diff
@@ -59,30 +60,50 @@ public sealed class IoTSdkFactory : IClientFactory, IDisposable {
             var ehubHost = Environment.GetEnvironmentVariable(IoTEdgeVariables.IOTEDGE_GATEWAYHOSTNAME);
 
             try {
-                if (!string.IsNullOrEmpty(config.EdgeHubConnectionString)) {
-                    _cs = IotHubConnectionStringBuilder.Create(config.EdgeHubConnectionString);
+                if (!string.IsNullOrEmpty(config.MqttClientConnectionString) &&
+                    !string.IsNullOrEmpty(config.EdgeHubConnectionString)) {
+                    throw new InvalidConfigurationException(
+                        "Can't have both a mqtt client connection string and a device connection string.");
+                }
+
+                if (!string.IsNullOrEmpty(config.MqttClientConnectionString)) {
+                    _mqttClientCs = MqttClientConnectionStringBuilder.Create(config.MqttClientConnectionString);
 
-                    if (string.IsNullOrEmpty(_cs.SharedAccessKey)) {
+                    if (_mqttClientCs.UsingIoTHub && string.IsNullOrEmpty(_mqttClientCs.SharedAccessSignature)) {
                         throw new InvalidConfigurationException(
                             "Connection string is missing shared access key.");
                     }
-                    if (string.IsNullOrEmpty(_cs.DeviceId)) {
+                    if (string.IsNullOrEmpty(_mqttClientCs.DeviceId)) {
                         throw new InvalidConfigurationException(
                             "Connection string is missing device id.");
                     }
 
-                    deviceId = _cs.DeviceId;
-                    moduleId = _cs.ModuleId;
-                    ehubHost = _cs.GatewayHostName ?? ehubHost;
+                    deviceId = _mqttClientCs.DeviceId;
+                    moduleId = _mqttClientCs.ModuleId;
```

Comment:
is there a gatewayHostName in the mqtt connection string ghat we're missing here?

Reply:
No, the broker host name is the target server to talk to.